### DISCUSSION TITLE
Sensible way to handle "patch chars" to sysex

### DISCUSSION THREAD

**Discussion by shooking** [RomanKubiak/ctrlr#184] [General]
Hi all

I hope we can open up the wiki? In the meantime I have a Zoom pedal.
It has a range of chars that can be set.
What's the best way to create a map so that the char can appear in a combo box say, and when you choose the index this is used to generate the value Zoom wants .. ie A = 65 is standard. But the order of the Zoom display is some circular buffer like:

A ==65
B == 66 etc etc
C
D
E
F
G
H
I
J
K
L
M
N
O
P
Q
R
S
T
U
V
W
X
Y
Z
a
b
c
d
e
f
g
h
i
j
k
l
m
n
o
p
q
r
s
t
u
v
w
x
y
z
0
1
2
3
4
5
6
7
8
9
!
#
$
%
&
'
(
)
+
,
-
.
:
=
@
[
]
^
_
`
{
}
~

**Comment by Ted-juh** [comment]
Looks like ASCII to me. 
![ASCII](https://user-images.githubusercontent.com/71118185/108641198-466aef00-749e-11eb-9935-0cb0560b42ed.png)

**Comment by shooking** [comment]
Super cool spot :-). I am so old I would have hoped I would have seen this
-). So I start ar space and add offset to the combo index.
Thanks.

Would be cool to open up a wiki for sure.

**Comment by Godlike-Productions** [comment]
I use this

```
initializePanel = function()

symbols = {	[0]="","|","|","|","|","|","|","|","|","|","|","|","|","|","|","|",
				"|","|","|","|","|","|","|","|","|","|","|","|","|","|","|","|",
				" ","!","\"","#","$","%","&","`","(",")","*","+",",","-",".","/",
				"0","1","2","3","4","5","6","7","8","9",":",";","<","=",">","?",
				"@","A","B","C","D","E","F","G","H","I","J","K","L","M","N","O",
				"P","Q","R","S","T","U","V","W","X","Y","Z","[","\\","]","^","_",
				"`","a","b","c","d","e","f","g","h","i","j","k","l","m","n","o",
				"p","q","r","s","t","u","v","w","x","y","z","{","|","}","~","|"}
	symbols2 = {	[0]="","|","|","|","|","|","|","|","|","|","|","|","|","|","|","|",
				"|","|","|","|","|","|","|","|","|","|","|","|","|","|","|","|",
				" ","!","\"","#","$","%","&","\'","(",")","*","+",",","-",".","/",
				"0","1","2","3","4","5","6","7","8","9",":",";","<","=",">","?",
				"@","A","B","C","D","E","F","G","H","I","J","K","L","M","N","O",
				"P","Q","R","S","T","U","V","W","X","Y","Z","[","\\","]","^","_",
				"`","a","b","c","d","e","f","g","h","i","j","k","l","m","n","o",
				"p","q","r","s","t","u","v","w","x","y","z","{","|","}","~","|"}
```

Can't remember where I got it from, probably Miniak
